AnaProd/HH_bbtautau/anaTupleDef.py

```diff
@@ -83,20 +83,42 @@ def getDefaultColumnsToSave(isData):
 def addAllVariables(dfw, syst_name, isData, trigger_class, lepton_legs, isSignal, global_params):
     dfw.Apply(CommonBaseline.SelectRecoP4, syst_name, global_params["nano_version"])
     dfw.Apply(AnaBaseline.RecoHttCandidateSelection, global_params)
-    dfw.Apply(AnaBaseline.RecoJetSelection)
+    dfw.Apply(AnaBaseline.RecoJetSelection, global_params["era"])
     dfw.Apply(AnaBaseline.ThirdLeptonVeto)
-    dfw.Apply(AnaBaseline.DefineHbbCand)
+    dfw.Apply(AnaBaseline.DefineHbbCand, global_params["met_type"])
     dfw.DefineAndAppend("Hbb_isValid" , "HbbCandidate.has_value()")
-    dfw.Apply(AnaBaseline.ExtraRecoJetSelection)
-    dfw.Apply(Corrections.getGlobal().jet.getEnergyResolution)
-    dfw.Apply(Corrections.getGlobal().btag.getWPid)
+    dfw.Apply(AnaBaseline.ExtraRecoJetSelection, global_params["era"])
+    if global_params["era"].startswith("Run2"):
+        dfw.Apply(Corrections.getGlobal().jet.getEnergyResolution)
+        dfw.Apply(Corrections.getGlobal().btag.getWPid)
     jet_obs = []
     jet_obs.extend(JetObservables)
     dfw.Apply(AnaBaseline.ApplyJetSelection)
     if not isData:
-        dfw.Define(f"Jet_genJet_idx", f" FindMatching(Jet_p4,GenJet_p4,0.3)")
+        #dfw.Define(f"Jet_genJet_idx", " FindMatching(Jet_p4,GenJet_p4,0.3)")
+        dfw.Define(f"Jet_genJet_idx", "Jet_genJetIdx")
```

Comment:
- Switch to `Take` (once difference in behaviours understood)
- Don't need to define `Jet_genJet_idx` - use `Jet_genJetIdx` directly